<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Samtools-Manual-CN by CNCBI</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Samtools-Manual-CN</h1>
      <h2 class="project-tagline">Samtools&#39; manual in Chinese - Samtools中文使用手册</h2>
      <a href="http://www.htslib.org/doc/samtools.html" class="btn">View in English</a>
      <a href="https://github.com/CNCBI/Samtools-Manual-CN" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">

<div class="container manpage">
<h1>NAME - 名字</h1>
samtools – 序列比对/图谱（SAM）格式的工具箱
<h1>SYNOPSIS - 概要</h1>
<p>
samtools view -bt ref_list.txt -o aln.bam aln.sam.gz
</p><p>
samtools sort -T /tmp/aln.sorted -o aln.sorted.bam aln.bam
</p><p>
samtools index aln.sorted.bam
</p><p>
samtools idxstats aln.sorted.bam
</p><p>
samtools view aln.sorted.bam chr2:20,100,000-20,200,000
</p><p>
samtools merge out.bam in1.bam in2.bam in3.bam
</p><p>
samtools faidx ref.fasta
</p><p>
samtools fixmate in.namesorted.sam out.bam
</p><p>
samtools mpileup -C50 -gf ref.fasta -r chr3:1,000-2,000 in1.bam in2.bam
</p><p>
samtools tview aln.sorted.bam ref.fasta
</p><p>
samtools flags PAIRED,UNMAP,MUNMAP
</p><p>
samtools bam2fq input.bam &gt; output.fastq
</p><p>
</p><h1>DESCRIPTION - 描述</h1>
<p>
Samtools是一个用来处理BAM格式（SAM的二进制格式，译者注）的比对文件的工具箱。它能够输入和输出SAM（序列比对/图谱）格式的文件，对其进行排序、合并、建立索引等处理，还可以快速获取任意区域的reads。
</p><p>
Samtools被设计成以数据流的形式工作。它把标准输入作为输入（stdin）文件 `-' ，把标准输出作为输出（stdout）文件 `-' 。因此多个命令可以通过Unix的管道技术结合起来。Samtools总是把警告和错误消息输出到标准错误输出（stderr）。
</p><p>
Samtools也能够打开一个远程FTP或HTTP服务器上的BAM（SAM不行）文件，但是这个BAM文件名要以 `ftp://' 或 `http://' 开头。Samtools会在当前工作目录下检查相应的索引文件，如果不存在的话，它会把它下载下来。Samtools不会获取整个比对文件，除非你要求它这么做。
</p><p>
</p><h1>COMMANDS AND OPTIONS - 命令和参数</h1>
<p>
</p><dl><dt><b>view</b></dt><dd><p>
samtools view
[<em>options</em>]
<em>in.bam</em>|<em>in.sam</em>|<em>in.cram</em>
[<em>region</em>...]
</p><p>
如果没有指定参数或者区域，这条命令会以SAM格式（不含头文件）打印输入文件（SAM，BAM或CRAM格式）里的所有比对到标准输出。
</p><p>
你可以在输入文件的文件名后面指定一个或多个以空格分隔的区域来限制输出，这样只会输出落在指定区域内的比对。要指定区域，需要一个对坐标排过序的并且建立了索引的输入文件（BAM或CRAM格式）。
</p><p>
<b>-b</b>,
<b>-C</b>,
<b>-1</b>,
<b>-u</b>,
<b>-h</b>,
<b>-H</b>,
和
<b>-c</b>
参数会改变默认的没有头文件的SAM输出格式。
<b>-o</b>
<b>-U</b>
参数可以设置输出文件的名字。
</p><p>
<b>-t</b>
和
<b>-T</b>
参数用来提供额外的参考数据，当输入的SAM没有@SQ头文件的时候，必须需要提供这两个中的一个参数。无论何时，只要是输出CRAM格式，就必须提供
<b>-T</b>
参数。
</p><p>
<b>-L</b>,
<b>-r</b>,
<b>-R</b>,
<b>-q</b>,
<b>-l</b>,
<b>-m</b>,
<b>-f</b>,
和
<b>-F</b>
参数用来过滤比对，只有那些满足特定标准的比对会被输出。
</p><p>
<b>-x</b>,
<b>-B</b>,
和
<b>-s</b>
参数用来修饰每个比对中的数据。
</p><p>
最后
<b>-@</b>
参数可用来为压缩分配额外的线程，而
<b>-?</b>
参数可以得到一个很长的帮助信息。
</p><p>
</p></dd><dt><b>REGIONS - 区域:</b></dt><dd><p>
区域可以用如下方法制定：RNAME[:STARTPOS[-ENDPOS]] ，即：染色体名[:起始位点[-终止位点]]，（[ ] 中的内容表示为可选内容，可以指定，也可以不指定。译者注），并且，所有的坐标系统都是以1标记第一个碱基的（1-based）。
</p><p>
要非常注意的是：如果给定了多个区域，有些比对可能会被输出多次，因为它们和多个区域都匹配。
</p><p>
指定区域的例子:
</p><dl><dt>`chr1'</dt><dd><p>
把所有比对到参考序列中名为`chr1'（也就是头文件里标明的 @SQ SN:chr1）的比对全部输出。（chr一般为染色体的缩写，译者注）
</p></dd><dt>`chr2:1000000'</dt><dd><p>
chr2上从第1,000,000碱基开始的序列到这条染色体的结尾的区域。
</p></dd><dt>`chr3:1000-2000'</dt><dd><p>
chr3上从第1,000个碱基到第2,000个碱基之间的共1001bp的区域（包含两个端点）。
</p></dd></dl>
<p>
<b>OPTIONS - 参数:</b>
</p><dl><dt><b>-b</b></dt><dd><p>
以BAM格式输出。
</p></dd><dt><b>-C</b></dt><dd><p>
以CRAM格式输出（要求有 -T 参数）。
</p></dd><dt><b>-1</b></dt><dd><p>
启用快速BAM压缩（隐含 -b）。
</p></dd><dt><b>-u</b></dt><dd><p>
输出未压缩的BAM。这个参数省下了压缩/解压缩的时间，因此更适合在运用管道技术把另一个samtools命令结合在一起时使用。
</p></dd><dt><b>-h</b></dt><dd><p>
在输出中包含头文件。
</p></dd><dt><b>-H</b></dt><dd><p>
只输出头文件。
</p></dd><dt><b>-c</b></dt><dd><p>
与输出比对不同，只对它们计数，并且输出总数。所有的过滤参数，比如
<b>-f</b>,
<b>-F</b>,
和
<b>-q</b>,
都可以被考虑进来。
</p></dd><dt><b>-?</b></dt><dd><p>
输出长篇帮助文档并且立即退出。
</p></dd><dt><b>-o </b><em>FILE</em></dt><dd><p>
输出到
<em>FILE</em>
[stdout] （[ ]中表示默认值，这里表示如果使用了-o参数，但没有提供文件名，输出到标准输出。译者注）。
</p></dd><dt><b>-U </b><em>FILE</em></dt><dd><p>
把那些<em>没有</em>被各种过滤参数所选中的比对，输出到
<em>FILE</em>里面。
当这个参数被使用是，所有的比对（如果指定了区域，则是所有和指定区域相交的比对）要么被写进输出文件，要么被写进这个文件，但绝不会同时被写进两个文件。
</p></dd><dt><b>-t </b><em>FILE</em></dt><dd><p>
一个以制表符（tab）分隔的文件
<em>FILE</em>。
每一行必须含有参考序列的名字，放在第一列，和这个参考序列的长度，放在第二列，每一行代表一个不同的参考序列。第二列后面的任何字段都会被忽略。这个文件同时也定义了在排序时的参考序列的顺序。如果你运行`samtools faidx &lt;ref.fa&gt;'，那么它输出的索引文件
<em>&lt;ref.fa&gt;.fai</em>
可以被用作这里的
<em>FILE</em>。
</p></dd><dt><b>-T </b><em>FILE</em></dt><dd><p>
一个FASTA格式的参考序列文件
<em>FILE</em>，可以是被
<b>bgzip</b>
压缩过的，并且最好是被
<b>samtools faidx</b>建立过索引的。
如果索引文件不存在，我们会为你生成一个。
</p></dd><dt><b>-L </b><em>FILE</em></dt><dd><p>
只输出那些和输入的BED<em>FILE</em>重合的比对 [null]（默认为空）。
</p></dd><dt><b>-r </b><em>STR</em></dt><dd><p>
只输出 <em>STR</em> 这个read group里面的比对 [null]。
</p></dd><dt><b>-R </b><em>FILE</em></dt><dd><p>
输出文件 <em>FILE</em> 里面列出来的read groups里的比对 [null]。
</p></dd><dt><b>-q </b><em>INT</em></dt><dd><p>
当比对的MAPQ小于 <em>INT</em> 时，忽略这个比对 [0]。
</p></dd><dt><b>-l </b><em>STR</em></dt><dd><p>
只输出 <em>STR</em> 这个文库（library）里面的比对 [null]。
</p></dd><dt><b>-m </b><em>INT</em></dt><dd><p>
Only output alignments with number of CIGAR bases consuming query
sequence ≥
<em>INT</em>
[0]
</p></dd><dt><b>-f </b><em>INT</em></dt><dd><p>
Only output alignments with all bits set in
<em>INT</em>
present in the FLAG field.
<em>INT</em>
can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/)
or in octal by beginning with `0' (i.e. /^0[0-7]+/) [0].
</p></dd><dt><b>-F </b><em>INT</em></dt><dd><p>
Do not output alignments with any bits set in
<em>INT</em>
present in the FLAG field.
<em>INT</em>
can be specified in hex by beginning with `0x' (i.e. /^0x[0-9A-F]+/)
or in octal by beginning with `0' (i.e. /^0[0-7]+/) [0].
</p></dd><dt><b>-x </b><em>STR</em></dt><dd><p>
Read tag to exclude from output (repeatable) [null]
</p></dd><dt><b>-B</b></dt><dd><p>
Collapse the backward CIGAR operation.
</p></dd><dt><b>-s </b><em>FLOAT</em></dt><dd><p>
Integer part is used to seed the random number generator [0]. Part after the
decimal point sets the fraction of templates/pairs to subsample [no subsampling].
</p></dd><dt><b>-@ </b><em>INT</em></dt><dd><p>
进行BAM压缩操作时，除了主线程之外，额外分配的线程数 [0]。
</p></dd><dt><b>-S</b></dt><dd><p>
这个参数会被忽略，它只是为了和旧版本的samtools相兼容。之前，如果输入文件是SAM格式的话，必须添加这个参数，但是现在，软件会侦测输入的前几个字符，自动确定正确的格式。
</p></dd></dl>
<p>
</p></dd><dt><b>tview</b></dt><dd><p>
samtools tview
[<b>-p</b>
<em>chr:pos</em>]
[<b>-s</b>
<em>STR</em>]
[<b>-d</b>
<em>display</em>]
&lt;in.sorted.bam&gt;
[ref.fasta]
</p><p>
文字版的reads和参考序列的比对情况的查看器（基于 ncurses library ）。在这个查看器中，按下 `?' 来寻求帮助，按下 `g' 可以从某个区域开始查看比对，这个区域用如下格式来表示：`chr10:10,000,000'，或者，当继续查看相同的参考序列时，可以使用 `=10,000,000'。
</p><p>
<b>Options:</b>
</p><dl><dt><b>-d</b><em> display</em></dt><dd><p>
以Html格式（H）、Curses格式（C）或文本格式（T）输出。
</p></dd><dt><b>-p</b><em> chr:pos</em></dt><dd><p>
直接到达这个位置。
</p></dd><dt><b>-s</b><em> STR</em></dt><dd><p>
只显示 <em> STR</em> 这个样本或read group的比对。
</p></dd></dl>
<p>
</p></dd><dt><b>mpileup</b></dt><dd><p>
samtools mpileup
[<b>-EBugp</b>]
[<b>-C</b>
<em>capQcoef</em>]
[<b>-r</b>
<em>reg</em>]
[<b>-f</b>
<em>in.fa</em>]
[<b>-l</b>
<em>list</em>]
[<b>-Q</b>
<em>minBaseQ</em>]
[<b>-q</b>
<em>minMapQ</em>]
<em>in.bam</em>
[<em>in2.bam</em>
[<em>...</em>]]
</p><p>
对一个或多个BAM文件，生成VCF、BCF或者pileup（堆积文件）。比对记录会按照头文件中@RG行里面样本标识符（SM）来分组。如果样本标识符不存在，那么每一个输入文件都被视作一个独立的样本。
</p><p>
在（没有使用 <b>-u</b> 或 <b> -g</b> 参数的）pileup格式中，每一行代表基因组上的一个位置，它由染色体名字，1-based坐标（该染色体的第一个碱基标为1），参考序列上的碱基，覆盖这个位点的reads数目，read上的碱基，碱基的质量和比对的质量构成。下列信息：匹配、不匹配、插入删除、正反向链、比对质量、是否是某read的起始或终止碱基，全部都编码在read上的碱基这里面了。在这一列中，一个点代表和参考序列正向链相同的一个匹配，一个逗号代表和反向链相同的一个匹配。一个 '&gt;' 或 '&lt;' 代表a reference skip（抱歉，译者还不确定含义，不知该如何翻译），`ACGTN'代表正向链上的一个mismatch（不匹配），`acgtn'代表反向链上的一个mismatch。这样的一个样式，`\\+[0-9]+[ACGTNacgtn]+'，代表了在参考序列上这个位置和下个位置之间的一个insertion（插入）。Insertion的长度由样式中的数字给出，它后面是插入的序列。同样的，这个样式，`-[0-9]+[ACGTNacgtn]+'，代表一个deletion（删除）。在之后的行中，被删除的碱基会用 `*' 来代表。同样在这一列中，一个 `^' 符号表示是某read的起始点。 `^'的ASCII码减去33给出了比对质量。一个 `$' 表示是某read的终止点。
</p><p>
<b>Input Options - 输入参数:</b>
</p><dl><dt><b>-6, --illumina1.3+</b></dt><dd><p>
假设质量是用Illumina 1.3+编码的话。
</p></dd><dt><b>-A, --count-orphans</b></dt><dd><p>
在识别变异时，不要跳过反常的read。
</p></dd><dt><b>-b, --bam-list</b><em> FILE</em></dt><dd><p>
输入的BAM文件的地址的列表，一个文件一行。 [null]
</p></dd><dt><b>-B, --no-BAQ</b></dt><dd><p>
Disable probabilistic realignment for the computation of base alignment
quality (BAQ). BAQ is the Phred-scaled probability of a read base being
misaligned. Applying this option greatly helps to reduce false SNPs
caused by misalignments.
</p></dd><dt><b>-C, --adjust-MQ</b><em> INT</em></dt><dd><p>
Coefficient for downgrading mapping quality for reads containing
excessive mismatches. Given a read with a phred-scaled probability q of
being generated from the mapped position, the new mapping quality is
about sqrt((INT-q)/INT)*INT. A zero value disables this
functionality; if enabled, the recommended value for BWA is 50. [0]
</p></dd><dt><b>-d, --max-depth</b><em> INT</em></dt><dd><p>
在一个位置上，对每个BAM输入文件，最多读取 <em>INT</em> 个reads。 [250]
</p></dd><dt><b>-E, --redo-BAQ</b></dt><dd><p>
在运行时重新计算BAQ，忽略存在的BQ标签。
</p></dd><dt><b>-f, --fasta-ref</b><em> FILE</em></dt><dd><p>
用<b>faidx</b>建立了索引的FASTA格式的参考序列文件，这个文件也可以是被<b>bgzip</b>压缩过的。 [null]
</p></dd><dt><b>-G, --exclude-RG</b><em> FILE</em></dt><dd><p>
排除列在文件FILE里面的readgroups里面的reads。（每一个@RG-ID占一行）
</p></dd><dt><b>-l, --positions</b><em> FILE</em></dt><dd><p>
BED或者是列了位置坐标的文件，这个文件含有一列区间或位点，这些地方是生产pileup或BCF的位置。如果是BED文件，坐标系统是0-based half-open（第一个碱基是0，区间是半开的）的。[null]
</p></dd><dt><b>-q, -min-MQ</b><em> INT</em></dt><dd><p>
最小的比对质量，某个比对的质量超过该值时才会被使用。[0]
</p></dd><dt><b>-Q, --min-BQ</b><em> INT</em></dt><dd><p>
最小的碱基质量，某个碱基的质量超过该值时才会被使用。[13]
</p></dd><dt><b>-r, --region</b><em> STR</em></dt><dd><p>
只产生这个区域的pileup。需要建立了索引的BAM文件。如果和 -l 同时使用的话，会考虑两个要求的交集。<em>STR</em>[all sites]
</p></dd><dt><b>-R, --ignore-RG</b></dt><dd><p>
忽略RG标签，把一个BAM文件中所有的reads都视为一个样本。
</p></dd><dt><b>--rf, --incl-flags</b><em> STR|INT</em></dt><dd><p>
需要的flag的标签：不在这里的mask bits（flag的值）的reads会被忽略 [null]
</p></dd><dt><b>--ff, --excl-flags</b><em> STR|INT</em></dt><dd><p>
过滤flags：忽略有这些mask bits标记的reads [UNMAP,SECONDARY,QCFAIL,DUP]
</p></dd><dt><b>-x, --ignore-overlaps</b></dt><dd><p>
取消read-pair（双端测序的两个reads）的重叠的检查。
</p></dd></dl><p>
<b>Output Options - 输出参数:</b>
</p><dl><dt><b>-o, --output </b><em>FILE</em></dt><dd><p>
把pileup或者VCF/BCF输出到<em>FILE</em>，而不是默认的标准输出里面。
</p><p>
（这个短参数同时被 <b>--open-prob</b> 和 <b>--output</b> 使用。如果<b>-o</b>的参数含有任何非数字的字符，而不是由 + 或 - 符号开头，它会被视作是<b>--output</b>。通常文件的扩展名会考虑到这一点，但是如果要写入到一个完全由数字组成的文件名，使用 <b>-o ./123</b> 或 <b>--output 123</b> 。）
</p></dd><dt><b>-g, --BCF</b></dt><dd><p>
计算基因型的概率，并输出到binary call format（BCF，二进制识别格式）。在v1.0版本的samtools中，它是BCF2，和BCF1格式是不兼容的，而以前的版本（0.1.x）输出的是BCF1。
</p></dd><dt><b>-v, --VCF</b></dt><dd><p>
计算基因型的概率，并输出到variant call format（VCF，变异识别格式）。输出是bgzip压缩后的VCF，除非设定了<b>-u</b>参数。
</p></dd></dl><p>
<b>Output Options for mpileup format (without -g or -v) - mpileup格式的输出参数（没有 -g 或 -v）:</b>
</p><dl><dt><b>-O, --output-BP</b></dt><dd><p>
输出reads里面碱基的位置。
</p></dd><dt><b>-s, --output-MQ</b></dt><dd><p>
输出mapping的质量。
</p></dd></dl><p>
<b>Output Options for VCF/BCF format (with -g or -v) - VCF/BCF格式的输出参数（有 -g 或 -v）:</b>
</p><dl><dt><b>-D</b></dt><dd><p>
输出每个样本的read depth（深度）[DEPRECATED - use
<b>-t DP</b>
instead]（不再宜用，使用<b>-t DP</b>来替代它）
</p></dd><dt><b>-S</b></dt><dd><p>
输出每个样本的Phred标准的的链的bias（偏差）的P-value（不再宜用，使用<b>-t SP</b>来替代它）
</p></dd><dt><b>-t, --output-tags</b><em> LIST</em></dt><dd><p>
输出逗号分隔的 FORMAT 和 INFO 标签的列表（大小写不敏感）：
<b>DP</b>
（高质量碱基的数目，FORMAT），
<b>DV</b>
（高质量的非参考序列的碱基数，FORMAT），
<b>DPR</b>
（支持每个观察到的allele（等位基因）的高质量的碱基数，FORMAT），
<b>INFO/DPR</b>
（支持每个观察到的allele（等位基因）的高质量的碱基数，INFO），
<b>DP4</b>
（高质量的 ref-forward, ref-reverse, alt-forward and alt-reverse （正向-参考序列，反向-参考序列，正向-突变序列，反向-突变序列）的碱基数目，FORMAT），
<b>SP</b>
（Phred标准的的链的bias（偏差）的P-value，FORMAT） [null]
</p></dd><dt><b>-u, --uncompressed</b></dt><dd><p>
输出未被压缩的VCF/BCF，在使用unix管道（pipe）时推荐这个参数。
</p></dd><dt><b>-V</b></dt><dd><p>
输出每个样本的非参考序列的reads数 （不再宜用，使用<b>-t DV</b>来替代它）
</p></dd></dl><p>
<b>Options for SNP/INDEL Genotype Likelihood Computation (for -g or -v):</b>
<b>对SNP/INDEL基因型的概率计算的参数 (for -g or -v):</b>
</p><dl><dt><b>-e, --ext-prob</b><em> INT</em></dt><dd><p>
Phred标准的gap延伸的测序错误概率。降低<em>INT</em>值会产生更长的indels。 [20]
</p></dd><dt><b>-F, --gap-frac</b><em> FLOAT</em></dt><dd><p>
最小的有gap的reads的比例。 [0.002]
</p></dd><dt><b>-h, --tandem-qual</b><em> INT</em></dt><dd><p>
对同聚物错误建模的系数。考虑一个长度为 <em>l</em> 的同聚物而言，大小为 <em>s</em> 的indel的测序错误是如下建模的：
<em>INT</em> * <em>s</em> / <em>l</em>.
[100]
</p></dd><dt><b>-I, --skip-indels</b></dt><dd><p>
不进行INDEL的识别。
</p></dd><dt><b>-L, --max-idepth</b><em> INT</em></dt><dd><p>
当每个样本的平均深度大于 <em>INT</em> 时，略过INDEL的识别。 [250]
</p></dd><dt><b>-m, --min-ireads</b><em> INT</em></dt><dd><p>
识别候选indel时，最小的存在gap的reads数目 <em>INT</em>。 [1]
</p></dd><dt><b>-o, --open-prob</b><em> INT</em></dt><dd><p>
由测序错误引入的gap的Phred-标准的概率。减少 <em>INT</em> 值会识别更多的indel。 [40]
</p><p>
（这个短参数同时被 <b>--open-prob</b> 和 <b>--output</b> 使用。当 <b>-o</b> 的参数是可选的 + 或 - 号，跟随着数字0-9的话，它会被视作 <b>--open-prob</b>。）
</p></dd><dt><b>-p, --per-sample-mF</b></dt><dd><p>

Apply
<b>-m</b>
and
<b>-F</b>
thresholds per sample to increase sensitivity of calling.
By default both options are applied to reads pooled from all samples.
</p></dd><dt><b>-P, --platforms</b><em> STR</em></dt><dd><p>
逗号分隔的一列仪器平台（由<b>@RG-PL</b>决定的），候选indel会从这些reads中获取。我们建议从低indel错误率的测序平台，比如ILLUMINA，中去收集候选indel。 [all]
</p></dd></dl>
<p>
</p></dd><dt><b>reheader</b></dt><dd><p>
samtools reheader &lt;in.header.sam&gt; &lt;in.bam&gt;
</p><p>
用 <em>in.header.sam</em> 里面的头文件替换 <em>in.bam</em> 里的。这个命令比通过BAM→SAM→BAM转换来替换头文件要快很多。
</p><p>
</p></dd><dt><b>cat</b></dt><dd><p>
samtools cat [-h header.sam] [-o out.bam] &lt;in1.bam&gt; &lt;in2.bam&gt; [ ... ]
</p><p>
Concatenate BAMs. The sequence dictionary of each input BAM must be identical,
although this command does not check this. This command uses a similar trick
to
<b>reheader</b>
which enables fast BAM concatenation.
</p><p>
</p></dd><dt><b>sort</b></dt><dd><p>
samtools sort
[<b>-l</b>
<em>level</em>]
[<b>-m</b>
<em>maxMem</em>]
[<b>-o</b>
<em>out.bam</em>]
[<b>-O</b>
<em>format</em>]
[<b>-n</b>]
<b>-T </b><em>out.prefix</em>
[<b>-@</b>
<em>threads</em>] [<em>in.bam</em>]
</p><p>
Sort alignments by leftmost coordinates, or by read name when
<b>-n</b>
is used.
An appropriate
<b>@HD-SO</b>
sort order header tag will be added or an existing one updated if necessary.
</p><p>
The sorted output is written to standard output by default, or to the
specified file
(<em>out.bam</em>)
when
<b>-o</b>
is used.
This command will also create temporary files
<em>out.prefix</em><b>.</b><em>%d</em><b>.bam</b>
as needed when the entire alignment data cannot fit into memory
(as controlled via the
<b>-m</b>
option).
</p><p>
<b>Options:</b>
</p><dl><dt><b>-l </b><em>INT</em></dt><dd><p>
Set the desired compression level for the final output file, ranging from 0
(uncompressed) or 1 (fastest but minimal compression) to 9 (best compression
but slowest to write), similarly to
<b>gzip</b>(1)'s
compression level setting.
</p><p>
If
<b>-l</b>
is not used, the default compression level will apply.
</p></dd><dt><b>-m </b><em>INT</em></dt><dd><p>
Approximately the maximum required memory per thread, specified either in bytes
or with a
<b>K</b>, <b>M</b>, or <b>G</b>
suffix.
[768 MiB]
</p></dd><dt><b>-n</b></dt><dd><p>
Sort by read names (i.e., the
<b>QNAME</b>
field) rather than by chromosomal coordinates.
</p></dd><dt><b>-o </b><em>FILE</em></dt><dd><p>
Write the final sorted output to
<em>FILE</em>,
rather than to standard output.
</p></dd><dt><b>-O </b><em>FORMAT</em></dt><dd><p>
Write the final output as
<b>sam</b>, <b>bam</b>, or <b>cram</b>.
</p><p>
By default, samtools tries to select a format based on the
<b>-o</b>
filename extension; if output is to standard output or no format can be
deduced,
<b>-O</b>
must be used.
</p></dd><dt><b>-T </b><em>PREFIX</em></dt><dd><p>
Write temporary files to
<em>PREFIX</em><b>.</b><em>nnnn</em><b>.bam.</b>
This option is required.
</p></dd><dt><b>-@ </b><em>INT</em></dt><dd><p>
Set number of sorting and compression threads.
By default, operation is single-threaded.
</p></dd></dl><p>
For compatibility with existing scripts,
<b>samtools sort</b>
also accepts the previous less flexible way of specifying the final and
temporary output filenames:
</p><p>
samtools sort
[<b>-nof</b>] [<b>-m</b>
<em>maxMem</em>]
<em>in.bam out.prefix</em>
</p><p>
The sorted BAM output is written to
<em>out.prefix</em><b>.bam</b>
(or as determined by the
<b>-o</b>
and
<b>-f</b>
options below) and any temporary files are written alongside as
<em>out.prefix</em><b>.</b><em>%d</em><b>.bam.</b>
</p><p>
</p><dl><dt><b>-o</b></dt><dd><p>
Output the final alignment to the standard output.
</p></dd><dt><b>-f</b></dt><dd><p>
Use
<em>out.prefix</em>
as the full output path and do not append
<b>.bam</b>
suffix.
</p></dd><dt><b>-l</b>, <b>-m</b>, <b>-n</b>, <b>-@</b></dt><dd><p>
Accepted with the same meanings as above.
</p></dd></dl><p>
This will eventually be removed; you should move to using the more flexible
newer style of invocation.
</p><p>
</p></dd><dt><b>merge</b></dt><dd><p>
samtools merge [-nur1f] [-h inh.sam] [-R reg] [-b &lt;list&gt;] &lt;out.bam&gt; &lt;in1.bam&gt; &lt;in2.bam&gt; [&lt;in3.bam&gt; ... &lt;inN.bam&gt;]
</p><p>
Merge multiple sorted alignment files, producing a single sorted output file
that contains all the input records and maintains the existing sort order.
</p><p>
If
<b>-h</b>
is specified the @SQ headers of input files will be merged into the specified header, otherwise they will be merged
into a composite header created from the input headers.  If in the process of merging @SQ lines for coordinate sorted
input files, a conflict arises as to the order (for example input1.bam has @SQ for a,b,c and input2.bam has b,a,c)
then the resulting output file will need to be re-sorted back into coordinate order.
</p><p>
Unless the
<b>-c</b>
or
<b>-p</b>
flags are specified then when merging @RG and @PG records into the output header then any IDs found to be duplicates
of existing IDs in the output header will have a suffix appended to them to diffientiate them from similar header
records from other files and the read records will be updated to reflect this.
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-1</b></dt><dd><p>
Use zlib compression level 1 to compress the output.
</p></dd><dt><b>-b</b><em> FILE</em></dt><dd><p>
List of input BAM files, one file per line.
</p></dd><dt><b>-f</b></dt><dd><p>
Force to overwrite the output file if present.
</p></dd><dt><b>-h</b><em> FILE</em></dt><dd><p>
Use the lines of
<em>FILE</em>
as `@' headers to be copied to
<em>out.bam</em>,
replacing any header lines that would otherwise be copied from
<em>in1.bam</em>.
(<em>FILE</em>
is actually in SAM format, though any alignment records it may contain
are ignored.)
</p></dd><dt><b>-n</b></dt><dd><p>
The input alignments are sorted by read names rather than by chromosomal
coordinates
</p></dd><dt><b>-R</b><em> STR</em></dt><dd><p>
Merge files in the specified region indicated by
<em>STR</em>
[null]
</p></dd><dt><b>-r</b></dt><dd><p>
Attach an RG tag to each alignment. The tag value is inferred from file names.
</p></dd><dt><b>-u</b></dt><dd><p>
Uncompressed BAM output
</p></dd><dt><b>-c</b></dt><dd><p>
Combine RG tags with colliding IDs rather than adding a suffix to differentiate them.
</p></dd><dt><b>-p</b></dt><dd><p>
Combine PG tags with colliding IDs rather than adding a suffix to differentiate them.
</p></dd></dl>
<p>
</p></dd><dt><b>index</b></dt><dd><p>
samtools index
[<b>-bc</b>]
[<b>-m</b>
<em>INT</em>]
<em>aln.bam</em>|<em>aln.cram</em>
</p><p>
Index a coordinate-sorted BAM or CRAM file for fast random access.
This index is needed when
<em>region</em>
arguments are used to limit
<b>samtools view</b>
and similar commands to particular regions of interest.
</p><p>
For a CRAM file
<em>aln.cram</em>,
index file
<em>aln.cram</em><b>.crai</b>
will be created.
For a BAM file
<em>aln.bam</em>,
either
<em>aln.bam</em><b>.bai</b>
or
<em>aln.bam</em><b>.csi</b>
will be created, depending on the index format selected.
</p><p>
<b>Options:</b>
</p><dl><dt><b>-b</b></dt><dd><p>
Create a BAI index.
This is currently the default when no format options are used.
</p></dd><dt><b>-c</b></dt><dd><p>
Create a CSI index.
By default, the minimum interval size for the index is 2^14, which is the same
as the fixed value used by the BAI format.
</p></dd><dt><b>-m </b><em>INT</em></dt><dd><p>
Create a CSI index, with a minimum interval size of 2^INT.
</p></dd></dl>
<p>
</p></dd><dt><b>idxstats</b></dt><dd><p>
samtools idxstats &lt;aln.bam&gt;
</p><p>
Retrieve and print stats in the index file. The output is TAB-delimited
with each line consisting of reference sequence name, sequence length, #
mapped reads and # unmapped reads.
</p><p>
</p></dd><dt><b>faidx</b></dt><dd><p>
samtools faidx &lt;ref.fasta&gt; [region1 [...]]
</p><p>
Index reference sequence in the FASTA format or extract subsequence from
indexed reference sequence. If no region is specified,
<b>faidx</b>
will index the file and create
<em>&lt;ref.fasta&gt;.fai</em>
on the disk. If regions are specified, the subsequences will be
retrieved and printed to stdout in the FASTA format. The input file can
be compressed in the
<b>BGZF</b>
format.
</p><p>
</p></dd><dt><b>fixmate</b></dt><dd><p>
samtools fixmate
[<b>-rpc</b>]
[<b>-O</b>
<em>format</em>]
<em>in.nameSrt.bam out.bam</em>
</p><p>
Fill in mate coordinates, ISIZE and mate related flags from a
name-sorted alignment.
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-r</b></dt><dd><p>
Remove secondary and unmapped reads.
</p></dd><dt><b>-p</b></dt><dd><p>
Disable FR proper pair check.
</p></dd><dt><b>-c</b></dt><dd><p>
Add template cigar ct tag.
</p></dd><dt><b>-O </b><em>FORMAT</em></dt><dd><p>
Write the final output as
<b>sam</b>, <b>bam</b>, or <b>cram</b>.
</p><p>
By default, samtools tries to select a format based on the output
filename extension; if output is to standard output or no format can be
deduced,
<b>-O</b>
must be used.
</p></dd></dl>
<p>
</p></dd><dt><b>rmdup</b></dt><dd><p>
samtools rmdup [-sS] &lt;input.srt.bam&gt; &lt;out.bam&gt;
</p><p>
Remove potential PCR duplicates: if multiple read pairs have identical
external coordinates, only retain the pair with highest mapping quality.
In the paired-end mode, this command
<b>ONLY</b>
works with FR orientation and requires ISIZE is correctly set. It does
not work for unpaired reads (e.g. two ends mapped to different
chromosomes or orphan reads).
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-s</b></dt><dd><p>
Remove duplicates for single-end reads. By default, the command works for
paired-end reads only.
</p></dd><dt><b>-S</b></dt><dd><p>
Treat paired-end reads and single-end reads.
</p></dd></dl>
<p>
</p></dd><dt><b>calmd</b></dt><dd><p>
samtools calmd [-EeubSr] [-C capQcoef] &lt;aln.bam&gt; &lt;ref.fasta&gt;
</p><p>
Generate the MD tag. If the MD tag is already present, this command will
give a warning if the MD tag generated is different from the existing
tag. Output SAM by default.
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-A</b></dt><dd><p>
When used jointly with
<b>-r</b>
this option overwrites the original base quality.
</p></dd><dt><b>-e</b></dt><dd><p>
Convert a the read base to = if it is identical to the aligned reference
base. Indel caller does not support the = bases at the moment.
</p></dd><dt><b>-u</b></dt><dd><p>
Output uncompressed BAM
</p></dd><dt><b>-b</b></dt><dd><p>
Output compressed BAM
</p></dd><dt><b>-S</b></dt><dd><p>
The input is SAM with header lines
</p></dd><dt><b>-C</b><em> INT</em></dt><dd><p>
Coefficient to cap mapping quality of poorly mapped reads. See the
<b>pileup</b>
command for details. [0]
</p></dd><dt><b>-r</b></dt><dd><p>
Compute the BQ tag (without -A) or cap base quality by BAQ (with -A).
</p></dd><dt><b>-E</b></dt><dd><p>
Extended BAQ calculation. This option trades specificity for sensitivity, though the
effect is minor.
</p></dd></dl>
<p>
</p></dd><dt><b>targetcut</b></dt><dd><p>
samtools targetcut [-Q minBaseQ] [-i inPenalty] [-0 em0] [-1 em1] [-2 em2] [-f ref] &lt;in.bam&gt;
</p><p>
This command identifies target regions by examining the continuity of read depth, computes
haploid consensus sequences of targets and outputs a SAM with each sequence corresponding
to a target. When option
<b>-f</b>
is in use, BAQ will be applied. This command is
<b>only</b>
designed for cutting fosmid clones from fosmid pool sequencing [Ref. Kitzman et al. (2010)].
</p><p>
</p></dd><dt><b>phase</b></dt><dd><p>
samtools phase [-AF] [-k len] [-b prefix] [-q minLOD] [-Q minBaseQ] &lt;in.bam&gt;
</p><p>
Call and phase heterozygous SNPs.
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-A</b></dt><dd><p>
Drop reads with ambiguous phase.
</p></dd><dt><b>-b</b><em> STR</em></dt><dd><p>
Prefix of BAM output. When this option is in use, phase-0 reads will be saved in file
<b>STR</b>.0.bam
and phase-1 reads in
<b>STR</b>.1.bam.
Phase unknown reads will be randomly allocated to one of the two files. Chimeric reads
with switch errors will be saved in
<b>STR</b>.chimeric.bam.
[null]
</p></dd><dt><b>-F</b></dt><dd><p>
Do not attempt to fix chimeric reads.
</p></dd><dt><b>-k</b><em> INT</em></dt><dd><p>
Maximum length for local phasing. [13]
</p></dd><dt><b>-q</b><em> INT</em></dt><dd><p>
Minimum Phred-scaled LOD to call a heterozygote. [40]
</p></dd><dt><b>-Q</b><em> INT</em></dt><dd><p>
Minimum base quality to be used in het calling. [13]
</p></dd></dl>
<p>
</p></dd><dt><b>flags</b></dt><dd><p>
samtools flags INT|STR[,...]
</p><p>
Convert between textual and numeric flag representation.
</p><p>
<b>FLAGS:</b>
</p><table>
<tbody><tr><td>0x1</td><td>PAIRED</td><td>.. paired-end (or multiple-segment) sequencing technology</td></tr>
<tr><td>0x2</td><td>PROPER_PAIR</td><td>.. each segment properly aligned according to the aligner</td></tr>
<tr><td>0x4</td><td>UNMAP</td><td>.. segment unmapped</td></tr>
<tr><td>0x8</td><td>MUNMAP</td><td>.. next segment in the template unmapped</td></tr>
<tr><td>0x10</td><td>REVERSE</td><td>.. SEQ is reverse complemented</td></tr>
<tr><td>0x20</td><td>MREVERSE</td><td>.. SEQ of the next segment in the template is reversed</td></tr>
<tr><td>0x40</td><td>READ1</td><td>.. the first segment in the template</td></tr>
<tr><td>0x80</td><td>READ2</td><td>.. the last segment in the template</td></tr>
<tr><td>0x100</td><td>SECONDARY</td><td>.. secondary alignment</td></tr>
<tr><td>0x200</td><td>QCFAIL</td><td>.. not passing quality controls</td></tr>
<tr><td>0x400</td><td>DUP</td><td>.. PCR or optical duplicate</td></tr>
<tr><td>0x800</td><td>SUPPLEMENTARY</td><td>.. supplementary alignment</td></tr>
</tbody></table>
<p>
</p></dd><dt><b>bam2fq</b></dt><dd><p>
samtools bam2fq [-nO] [-s &lt;outSE.fq&gt;] &lt;in.bam&gt;
</p><p>
Converts a bam into FASTQ format.
</p><p>
<b>OPTIONS:</b>
</p><dl><dt><b>-n</b></dt><dd><p>
By default, either '/1' or '/2' is added to the end of read names
where the corresponding BAM_READ1 or BAM_READ2 flag is set.
Using
<b>-n</b>
causes read names to be left as they are.
</p></dd><dt><b>-O</b></dt><dd><p>
Use quality values from OQ tags in preference to standard quality string
if available.
</p></dd><dt><b>-s FILE</b></dt><dd><p>
Write singleton reads in FASTQ format to FILE instead of outputting them.
</p></dd></dl>
<p>
</p><p>
</p></dd><dt><b>help</b>, <b>--help</b></dt><dd><p>
Display a brief usage message listing the samtools commands available.
If the name of a command is also given, e.g.,
<b>samtools help view</b>,
the detailed usage message for that particular command is displayed.
</p><p>
</p></dd><dt><b>--version</b></dt><dd><p>
Display the version numbers and copyright information for samtools and
the important libraries used by samtools.
</p><p>
</p></dd><dt><b>--version-only</b></dt><dd><p>
Display the full samtools version number in a machine-readable format.
</p></dd></dl><p>
</p><h1>REFERENCE SEQUENCES</h1>
<p>
The CRAM format requires use of a reference sequence for both reading
and writing.
</p><p>
When reading a CRAM the <b>@SQ</b> headers are interrogated to identify
the reference sequence MD5sum (<b>M5:</b> tag) and the local reference
sequence filename (<b>UR:</b> tag).  Note that <em>http://</em> and
<em>ftp://</em> based URLs in the UR: field are not used, but local fasta
filenames (with or without <em>file://</em>) can be used.
</p><p>
To create a CRAM the <b>@SQ</b> headers will also be read to identify
the reference sequences, but M5: and UR: tags may not be present. In
this case the <b>-T</b> and <b>-t</b> options of samtools view may be
used to specify the fasta or fasta.fai filenames respectively
(provided the .fasta.fai file is also backed up by a .fasta file).
</p><p>
The search order to obtain a reference is:
</p><p>
Use any local file specified by the command line options (eg -T).
</p><p>
Look for MD5 via REF_CACHE environment variable.
</p><p>
Look for MD5 in each element of the REF_PATH environment variable.
</p><p>
Look for a local file listed in the UR: header tag.
</p><p>
</p><h1>ENVIRONMENT VARIABLES</h1>
<p>
</p><dl><dt><b>REF_PATH</b></dt><dd><p>
A colon separated (semi-colon on Windows) list of locations in which
to look for sequences identified by their MD5sums.  This can be either
a list of directories or URLs. Note that if a URL is included then the
colon in http:// and ftp:// and the optional port number will be
treated as part of the URL and not a PATH field separator.
For URLs, the text <b>%s</b> will be replaced by the MD5sum being
read.
</p><p>
If no REF_PATH has been specified it will default to
<b>http://www.ebi.ac.uk/ena/cram/md5/%s</b> and if REF_CACHE is also unset,
it will be set to <b>$XDG_CACHE_HOME/hts-ref/%2s/%2s/%s</b>.
If <b>$XDG_CACHE_HOME</b> is unset, <b>$HOME/.cache</b> (or a local system
temporary directory if no home directory is found) will be used similarly.
</p><p>
</p></dd><dt><b>REF_CACHE</b></dt><dd><p>
This can be defined to a single directory housing a local cache of
references.  Upon downloading a reference it will be stored in the
location pointed to by REF_CACHE.  When reading a reference it will be
looked for in this directory before searching REF_PATH.  To avoid many
files being stored in the same directory, a pathname may be
constructed using %<em>num</em>s and %s notation, consuming <em>num</em>
characters of the MD5sum.  For example
<b>/local/ref_cache/%2s/%2s/%s</b> will create 2 nested subdirectories
with the filenames in the deepest directory being the last 28
characters of the md5sum.
</p><p>
The REF_CACHE directory will be searched for before attempting to load
via the REF_PATH search list.  If no REF_PATH is defined, both
REF_PATH and REF_CACHE will be automatically set (see above), but if
REF_PATH is defined and REF_CACHE not then no local cache is used.
</p><p>
To aid population of the REF_CACHE directory a script
<b>misc/seq_ref_populate.pl</b> is provided in the Samtools
distribution. This takes a fasta file or a directory of fasta files
and generates the MD5sum named files.
</p></dd></dl><p>
</p><h1>EXAMPLES</h1>
<p>
Import SAM to BAM when
<b>@SQ</b>
lines are present in the header:
</p><pre>samtools view -bS aln.sam &gt; aln.bam
</pre>
If
<b>@SQ</b>
lines are absent:
<pre>samtools faidx ref.fa
samtools view -bt ref.fa.fai aln.sam &gt; aln.bam
</pre>
where
<em>ref.fa.fai</em>
is generated automatically by the
<b>faidx</b>
command.
<p>
</p><p>
Convert a BAM file to a CRAM file using a local reference sequence.
</p><pre>samtools view -C -T ref.fa aln.bam &gt; aln.cram
</pre>
<p>
Attach the
<b>RG</b>
tag while merging sorted alignments:
</p><pre>perl -e 'print "@RG\\tID:ga\\tSM:hs\\tLB:ga\\tPL:Illumina\\n@RG\\tID:454\\tSM:hs\\tLB:454\\tPL:454\\n"' &gt; rg.txt
samtools merge -rh rg.txt merged.bam ga.bam 454.bam
</pre>
The value in a
<b>RG</b>
tag is determined by the file name the read is coming from. In this
example, in the
<em>merged.bam</em>,
reads from
<em>ga.bam</em>
will be attached
<em>RG:Z:ga</em>,
while reads from
<em>454.bam</em>
will be attached
<em>RG:Z:454</em>.
<p>
</p><p>
Call SNPs and short INDELs:
</p><pre>samtools mpileup -uf ref.fa aln.bam | bcftools call -mv &gt; var.raw.vcf
bcftools filter -s LowQual -e '%QUAL&lt;20 || DP&gt;100' var.raw.vcf  &gt; var.flt.vcf
</pre>
The
<b>bcftools filter</b>
command marks low quality sites and sites with the read depth exceeding
a limit, which should be adjusted to about twice the average read depth
(bigger read depths usually indicate problematic regions which are
often enriched for artefacts).  One may consider to add
<b>-C50</b>
to
<b>mpileup</b>
if mapping quality is overestimated for reads containing excessive
mismatches. Applying this option usually helps
<b>BWA-short</b>
but may not other mappers.
<p>
Individuals are identified from the
<b>SM</b>
tags in the
<b>@RG</b>
header lines. Individuals can be pooled in one alignment file; one
individual can also be separated into multiple files. The
<b>-P</b>
option specifies that indel candidates should be collected only from
read groups with the
<b>@RG-PL</b>
tag set to
<em>ILLUMINA</em>.
Collecting indel candidates from reads sequenced by an indel-prone
technology may affect the performance of indel calling.
</p><p>
</p><p>
Generate the consensus sequence for one diploid individual:
</p><pre>samtools mpileup -uf ref.fa aln.bam | bcftools call -c | vcfutils.pl vcf2fq &gt; cns.fq
</pre>
<p>
Phase one individual:
</p><pre>samtools calmd -AEur aln.bam ref.fa | samtools phase -b prefix - &gt; phase.out
</pre>
The
<b>calmd</b>
command is used to reduce false heterozygotes around INDELs.
<p>
</p><p>
</p><p>
Dump BAQ applied alignment for other SNP callers:
</p><pre>samtools calmd -bAr aln.bam &gt; aln.baq.bam
</pre>
It adds and corrects the
<b>NM</b>
and
<b>MD</b>
tags at the same time. The
<b>calmd</b>
command also comes with the
<b>-C</b>
option, the same as the one in
<b>pileup</b>
and
<b>mpileup</b>.
Apply if it helps.
<p>
</p><h1>LIMITATIONS</h1>
<p>
</p><p>
Unaligned words used in bam_import.c, bam_endian.h, bam.c and bam_aux.c.
</p><p>
Samtools paired-end rmdup does not work for unpaired reads (e.g. orphan
reads or ends mapped to different chromosomes). If this is a concern,
please use Picard's MarkDuplicates which correctly handles these cases,
although a little slower.
</p><p>
</p><h1>AUTHOR</h1>
<p>
Heng Li from the Sanger Institute wrote the C version of samtools. Bob
Handsaker from the Broad Institute implemented the BGZF library. John
Marshall and Petr Danecek contribute to the source code and various
people from the 1000 Genomes Project have contributed to the SAM format
specification.
</p><p>
</p><h1>SEE ALSO</h1>
<em>bcftools</em>(1),
<em>sam</em>(5)
<p>
Samtools website: &lt;http://samtools.sourceforge.net&gt;
<br>
Samtools latest source: &lt;https://github.com/samtools/samtools&gt;
<br>
File format specification of SAM/BAM,CRAM,VCF/BCF: &lt;http://samtools.github.io/hts-specs&gt;
<br>
HTSlib website: &lt;https://github.com/samtools/htslib&gt;
<br>
Bcftools website: &lt;http://samtools.github.io/bcftools&gt;
</p></div>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/CNCBI/Samtools-Manual-CN">Samtools-Manual-CN</a> is maintained by <a href="https://github.com/CNCBI">CNCBI</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

            <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-69782798-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
